pkg/ds/farm_test.go

```diff
@@ -791,7 +886,7 @@ func TestMultipleFarms(t *testing.T) {
 		dsToUpdate.NodeSelector = someSelector
 		return dsToUpdate, nil
 	}
-	_, err = dsStore.MutateDS(anotherDSData.ID, mutator)
+	anotherDSData, err = dsStore.MutateDS(anotherDSData.ID, mutator)
```

Comment:
i think this test was bogus!! before this line `anotherDSData` has selector `availability_zone=az2`. After it has `availability_zone=az3`. however we weren't reassigning to `anotherDSData` so our copy has `az2` still. then line 891 is testing that the farms' representations has `az3`.

This would succeed if you checked before the farms were able to update their in-memory copies of the daemon set, but fails if you're too slow!